extensions/expanded-opcodes/README.md

```diff
@@ -32,8 +32,15 @@ This extension adds an expanded instruction format to allow for a larger number
 | `0 0000 0000` | `ADC`           | `A ← A + B + C`                    | `ZNCV` | (2)         |
 | `0 0000 0001` | `SBB`           | `A ← A - B - C`                    | `ZNCV` | (2)         |
 | `0 0000 0010` | `RSBB`          | `A ← B - A - C`                    | `ZNCV` | (1) (2)     |
-| `0 0000 0011` |                 |                                    |        | reserved    |
-| `0 0000 01xx` |                 |                                    |        | reserved    |
+| `0 0000 0011` | `ASR`           | `A ← A >>> B`                      | `C`    | (3) (4)     |
+| `0 0000 0100` | `ROL`           | <code>A ← (A << B) &#124; (A >> -B)</code> | `C` | (3)   |
+| `0 0000 0101` | `ROR`           | <code>A ← (A >> B) &#124; (A << -B)</code> | `C` | (3)   |
+| `0 0000 0110` | `RCL`           | <code>C:A ← (C:A << B) &#124; (C:A >> -B)</code> | `C` | (2) (3) (5) |
+| `0 0000 0111` | `RCR`           | <code>A:C ← (A:C >> B) &#124; (A:C << -B)</code> | `C` | (2) (3) (5) |
+| `0 0000 1000` | `SHL`           | `A ← A << B`                       | `C`   | (3)         |
+| `0 0000 1001` | `SHR`           | `A ← A >> B`                       | `C`   | (3)         |
```

Comment:
1. Why are the zero and negative flags no longer being set? They seem like reasonable flags to set.
2. Why do the normal shifts and rotates set the carry flag? It seems like that would only be well defined for shifts by  1 (and not at all for normal rotates).

Reply:
1. Because other architectures don't do it. Specifically, when implementing OoO processors, it is shockingly useful to rename the flags register _at the bit level_. So the C flag is renamed separately from the V flag, which are renamed separately from the (usually combined) N and Z flags. This way, false flag-carried dependencies can be eliminated. Setting more flags = more dependencies, and checking Z and N after shifts (where the result being Z or N is rarely relevant) is rare. It can always be done by TESTing the result against itself when needed.
2. Every other architecture seems to do it. It's certainly well-defined for all shifts, and the comment explains how. Another way to see it is that the carry flag ends up with the same bit that it _would_ have ended up with if the operation were a rotate through carry - it just doesn't affect the operation otherwise.
I think it's useful for implementing floating point in software, but I'll have to actually do that before I can explain how.

Reply:
1. I'll need to do some of my own research on this, but it seems odd to not set the Z and N flags as well.
2. You do define a way to set carry for shifts and rotate with carry, but that doesn't explain what carry should be for ROL and ROR. Nothing is shifted out for those 2 instructions. Unless you're saying that they effectively become duplicates of the lowest bit?

Reply:
1. Alright. For the record I'm implementing this in my build and I'll see if it's useful in the program I want to write.
2. Yes, that's what it says.

Reply:
1. Not setting ZN seems odd to me as well, but I can't really think of a good use case. If we run into something, we could still change this. 
2. Could be mentioned explicitly, but not required IMO

Reply:
```suggestion
| `0 0000 0011` | `ASR`           | `A ← A >>> B`                      | `CZN`    | (3) (4)     |
| `0 0000 0100` | `ROL`           | <code>A ← (A << B) &#124; (A >> -B)</code> | `CZN` | (3)   |
| `0 0000 0101` | `ROR`           | <code>A ← (A >> B) &#124; (A << -B)</code> | `CZN` | (3)   |
| `0 0000 0110` | `RCL`           | <code>C:A ← (C:A << B) &#124; (C:A >> -B)</code> | `CZN` | (2) (3) (5) |
| `0 0000 0111` | `RCR`           | <code>A:C ← (A:C >> B) &#124; (A:C << -B)</code> | `CZN` | (2) (3) (5) |
| `0 0000 1000` | `SHL`           | `A ← A << B`                       | `CZN`   | (3)         |
| `0 0000 1001` | `SHR`           | `A ← A >> B`                       | `CZN`   | (3)         |
```

My justification for this suggestion is that, while Z and N may not be used, every other ALU instruction sets these flags so it would add functionality while also potentially simplifying hardware.